### DISCUSSION TITLE
API proxy not working

### DISCUSSION THREAD

**Discussion by jonmor51** [quasarframework/quasar#12584] [General - Components / Directives / etc]
I am working on a Quasar app which needs to talk to an API backend. On my dev machine the Quasar app is on port 8081 and the API on port 8080. To avoid CORS etc. issues, I want to set up a proxy, so requests to localhost:8080/api are proxied to the back end. Following the instructions in the docs, I have the following block in quasar.conf.js:
```
    devServer: {
       server: {
         type: "http",
       },
      port: 8081,
      open: true, // opens browser window automatically
      proxy: {
        // proxy all requests starting with /api to Grails
        '/api': {
          target: 'http://localhost:8080',
          changeOrigin: true,
           pathRewrite: {
             '^/api': ''
           }
        }
      }
    },
```
But this is not working. Requests to 'localhost:8081/api/' are getting a 404 response from Express. Do I have to do something else, either in quasar.conf.js or in the router configuration, to ensure that '/api/...' requests bypass the router and are proxied instead to the back end?

**Comment by yusufkandemir** [comment]
According to what you described, the proxying is working. Making HTTP requests isn't related to Vue Router at all, so it's not like it needs to bypass it. Since you are getting 404, your request is hitting your API server.
Does your API expect `http://localhost:8080/api/whatever` or `http://localhost:8080/whatever`? You can play around with `pathRewrite` to get it working, or simply enable logging on your API server to make sure it's receiving requests, and then analyze the incoming requests and why they are getting responded with 404, etc.

**Comment by jychen23333** [comment]
I had the same problem on quasar, the reason was because pathRewrite didn't work, changed to rewrite and it was fine, the reason was not clear
`proxy: {
        "/api": {
          target: "http://localhost:5000",
          changeOrigin: true,
          rewrite: (path) => path.replace(/^\/api/, ""),
        },
      }```

**Reply by yusufkandemir** [reply]
It's `rewrite` when using Quasar CLI with Vite as shown in the docs: https://quasar.dev/quasar-cli-vite/api-proxying#introduction
